### DISCUSSION TITLE
Equivalent of NHibernate batch_fetch_style property

### DISCUSSION THREAD

**Discussion by swhitlock** [nhibernate/fluent-nhibernate#483] [Q&A] [answered]
For my .HasMany() relationships I'm using something like .BatchSize(16) to alleviate N + 1 performance issues.  This works well, but a well-known problem is that it can significantly increase memory usage of BatchEntityLoader.  There is a parameter in NHibernate called [hibernate.batch_fetch_style](https://docs.jboss.org/hibernate/orm/4.2/manual/en-US/html/ch20.html#performance-fetching-batch) that lets you trade off a bit of performance for lower memory usage.  I would like to try the Dynamic setting.  How do I access this parameter in Fluent NHibernate?

**Comment by fredericDelaporte** [comment, marked as answer]
This parameter is not ported to NHibernate, you are referring to Hibernate documentation. So you cannot configure it with Fluent, since it does not exist in NHibernate.
The memory usage issue is known, see nhibernate/nhibernate-core#1316. Porting the Hibernate setting is one way to solve it, as already commented in the issue. It awaits someone interested in contributing a fix.

**Reply by swhitlock** [reply]
Ah, that makes more sense.  Thank you.
